lib/ash_authentication/strategies/magic_link/request.ex

```diff
@@ -44,7 +44,7 @@ defmodule AshAuthentication.Strategy.MagicLink.Request do
 
       {:ok, user} ->
         with {sender, send_opts} <- strategy.sender,
-             {:ok, token} <- MagicLink.request_token_for_identity(strategy, identity, context) do
+             {:ok, token} <- MagicLink.request_token_for(strategy, user, opts, context) do
```

Comment:
Lets put this back to the `request_token_for_identity`, but add an `opts` to that, which uses `Keyword.merge` like `request_token_for_identity` does 🙇

Reply:
And instead of using the `opts` from `run`, we need to pass in `Ash.Context.to_opts(context)` to that new `opts` argument. Make sure the new `opts` argument is optional for backwards compatibility, i.e `opts \\ []`.

Reply:
Alright, but regarding backwards compatibility this would mean `opts \\ []` would have to be at the end of the function, as last argument. While for `request_token_for` it's second to last. 

I tried doing it as `request_token_for` and changed everywhere where it's called but now I'm thinking this would be a breaking change for anyone using context as third argument.

That's why I only is putting it at the end, last argument; but that it's inconsistent with the first function so, that's but bugs me. 🐛

Reply:
It's fine if it's inconsistent. It is what it is 😆